cni-plugin/linkerd-cni-validator/src/main.rs

```diff
@@ -0,0 +1,169 @@
+
+    log_format.try_init(log_level)?;
+    let target_addr = SocketAddr::new(target_ip, target_port);
+
+    info!(%outbound_proxy_addr, %target_addr, "Validating outbound traffic is redirected to proxy's outbound port");
+
+    let (shutdown_tx, shutdown_rx) = kubert::shutdown::sigint_or_sigterm()?;
+    let (ready_tx, ready_rx) = tokio::sync::oneshot::channel::<Result<()>>();
+
+    let server_task = tokio::spawn(outbound_serve(outbound_proxy_addr, ready_tx, shutdown_rx));
+
+    let validation_task = tokio::spawn(validate_outbound_redirect(target_addr, ready_rx));
+
+    tokio::select! {
+        task = validation_task => {
+            let validation_result = task.expect("Failed to run validator task");
+            if let Err(err) = validation_result {
+                error!(error = %err, "Failed validation");
+                std::process::exit(UNSUCCESSFUL_EXIT_CODE);
+            }
+
+            info!("Validation passed successfully...exiting");
+            Ok(())
+        }
+
+        _ = shutdown_tx.signaled() => {
+            if let Err(e) = server_task.await.expect("Failed to run outbound server task") {
+                error!(error = %e, "Failed to validate outbound routing configuration");
+                std::process::exit(UNSUCCESSFUL_EXIT_CODE);
+            }
+
+            error!("Failed to validate due to server terminating early");
+            std::process::exit(UNSUCCESSFUL_EXIT_CODE);
+        }
+    }
+}
+
+async fn validate_outbound_redirect(
+    target_addr: SocketAddr,
+    ready_rx: tokio::sync::oneshot::Receiver<Result<()>>,
+) -> Result<()> {
+    let timeout = std::time::Duration::from_secs(120);
+    if (tokio::time::timeout(timeout, ready_rx).await).is_err() {
+        anyhow::bail!("timed-out ({:?}) waiting for server to be ready", timeout);
+    }
+
+    let mut stream = {
+        debug!(original_dst = %target_addr, "Building validation client");
+        let socket = TcpStream::connect(target_addr).await?;
+        let peer = socket.peer_addr()?;
+        debug!(original_dst = %peer, "Client connected to validation server");
+        socket
+    };
+
+    tokio::select! {
+        is_readable = stream.readable() => {
+            if let Err(e) = is_readable {
+                anyhow::bail!("cannot read off client socket {}", e);
+            }
+        }
+
+        () = tokio::time::sleep(timeout) => {
+            anyhow::bail!("timed-out ({:?}) waiting for socket to become readable", timeout);
+        }
+    };
+
+    let mut buf = [0u8; 100];
+    let read_sz = stream.read(&mut buf[..REDIRECT_RESPONSE.len()]).await?;
+    let resp = String::from_utf8(buf[..REDIRECT_RESPONSE.len()].to_vec())?;
+    debug!(redirect_response = %resp, bytes_read = %read_sz);
+
+    if resp == REDIRECT_RESPONSE {
+        Ok(())
+    } else {
+        anyhow::bail!(
+            "expected client to receive {:?}, got {:?} instead",
+            REDIRECT_RESPONSE,
+            resp,
+        );
+    }
```

Comment:
nit, take it or leave it: i think we could make this a bit simpler using [`anyhow`'s `ensure!` macro](https://docs.rs/anyhow/latest/anyhow/macro.ensure.html)
```suggestion
    anyhow::ensure!(
        resp == REDIRECT_RESPONSE,
        "expected client to receive {:?}, got {:?} instead",
        REDIRECT_RESPONSE,
        resp,
    );
    
    Ok(())
```